README
NAME
    ElkM1::Control - a module allowing control of the ElkM1 Security / Home Automation system.

    ElkM1::Control is a set of modules which provide an API for communication and control with the ElkM1 security and home automation system. This package contains no user-runnable code, but is an API with which you can build applications that can communicate with the Elk. With this package you can command the panel to do anything you could do at the keypad. You can also recieve messages when certain events occur on the panel such as a zone being violated, a task activating, etc. 

    This package requires the use of the Elk ethernet module. It is very possible that this could be easily ported to use the serial port. I'd welcome patches to make this happen.

    This system can use either the secure port (2601) or the insecure port (2101). See the use_ssl option in the new method of ElkM1::Control.  

PREREQUISITES 

    This package requires IO::Socket::INET and IO::Socket::SSL modules.

HOW TO BUILD 

    This package should be a fairly straight forward build. 

    perl Makefile.PL
    make
    make test

HOW TO INSTALL 

    make install

AUTHORS
    James Russo, "<jr@halo3.net>"

COPYRIGHT
    Copyright 2006 James Russo, All Rights Reserved.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.


Changes
Revision history for Perl extension ElkM1::Control.

0.1.3 2019-02-23
        - Switched SSLv3 to TLSv1 to support newer versions of SSL
        - Added SSL_verify_mode => 0 to disable new default of certificate verification
        - Fixed example usage in ZoneAnalogVoltageReply.pm
        - Fixed bug in turnOnPLCDevice (use of index caused croak)

0.1.2 2016-09-19
	- Fixed SSL connection authentication timeout issues caused by XEP firmware update 2.0.42       

0.1.1 2015-09-25
	- Fixed port configuration bug (would not actually use port from params)
	- Fixed SSL connections by setting to SSLv3
	- Added Username and Password authentication on SSH
	- Implemented CustomValueReply.pm

0.1.0 2009-01-29
	- added fix for PLCStatusReply, changed version to 0.1.0
	- replaced EthernetModuleTest with Taras Dejneka version

0.03  2009-01-18
	- fixed BypassedZoneState -> BypassedZoneStateReply
	- added EthernetModuleTest message

0.02  Thu Jan 19 08:27:44 EST 2006
    - initial alpha public release of ElkM1::Control. Basic
      function provided. More documentation, testing needed.

0.01  Mon Jan  9 14:02:59 2006
	- original version; created by h2xs 1.23 with options
		-AX ElkM1::Control



lib/ElkM1/Control/Words.pm
package ElkM1::Control::Words;

=head1 NAME

ElkM1::Control::Words - Module which contains a listing of all words available in the Elk

=cut

use strict;
use warnings;

our %ELK_M1_WORDS = (
 'custom1'            => 1,
 'custom2'            => 2,
 'custom3'            => 3,
 'custom4'            => 4,
 'custom5'            => 5,
 'custom6'            => 6,
 'custom7'            => 7,
 'custom8'            => 8,
 'custom9'            => 9,
 'custom10'           => 10,
 'zero'               => 21,
 'one'                => 22,
 'two'                => 23,
 'three'              => 24,
 'four'               => 25,
 'five'               => 26,
 'six'                => 27,
 'seven'              => 28,
 'eight'              => 29,
 'nine'               => 30,
 'ten'                => 31,
 'eleven'             => 32,
 'twelve'             => 33,
 'thirteen'           => 34,
 'fourteen'           => 35,
 'fifteen'            => 36,
 'sixteen'            => 37,
 'seventeen'          => 38,
 'eighteen'           => 39,
 'nineteen'           => 40,
 'twenty'             => 41,
 'thirty'             => 42,
 'fourty'             => 43,
 'fifty'              => 44,
 'sixty'              => 45,
 'seventy'            => 46,
 'eighty'             => 47,
 'ninety'             => 48,
 'hundred'            => 49,
 'thousand'           => 50,
 '[200ms_silence]'    => 51,
 '[500ms_silence]'    => 52,
 '[800hz_tone]'       => 53,
 'a'                  => 54,
 'access'             => 55,
 'acknowledged'       => 56,
 'ac_power'           => 57,
 'activate'           => 58,
 'activated'          => 59,
 'active'             => 60,
 'adjust'             => 61,
 'air'                => 62,
 'alarm'              => 63,
 'alert'              => 64,
 'all'                => 65,
 'am'                 => 66,
 'an'                 => 67,
 'and'                => 68,
 'answer'             => 69,
 'any'                => 70,
 'are'                => 71,
 'area'               => 72,
 'arm'                => 73,
 'armed'              => 74,
 'at'                 => 75,
 'attic'              => 76,
 'audio'              => 77,
 'auto'               => 78,
 'authorized'         => 79,
 'automatic'          => 80,
 'automation'         => 81,
 'auxiliary'          => 82,
 'away'               => 83,
 'b'                  => 84,
 'back'               => 85,
 'barn'               => 86,
 'basement'           => 87,
 'bathroom'           => 88,
 'battery'            => 89,
 'bedroom'            => 90,
 'been'               => 91,
 'bell'               => 92,
 'bottom'             => 93,
 'break'              => 94,
 'breakfast'          => 95,
 'bright'             => 96,
 'building'           => 97,
 'burglar'            => 98,
 'button'             => 99,
 'by'                 => 100,
 'bypassed'           => 101,
 'cabinet'            => 102,
 'call'               => 103,
 'camera'             => 104,
 'cancel'             => 105,
 'carbon_monoxide'    => 106,
 'card'               => 107,
 'center'             => 108,
 'central'            => 109,
 'change'             => 110,
 'check'              => 111,
 'chime'              => 112,
 'circuit'            => 113,
 'clear'              => 114,
 'closed'             => 115,
 'closet'             => 116,
 'code'               => 117,
 'cold'               => 118,
 'condition'          => 119,
 'connect'            => 120,
 'control'            => 121,
 'cool'               => 122,
 'cooling'            => 123,
 'corner'             => 124,
 'crawlspace'         => 125,
 'danger'             => 126,
 'day'                => 127,
 'deck'               => 128,
 'decrease'           => 129,
 'defective'          => 130,
 'degrees'            => 131,
 'delay'              => 132,
 'den'                => 133,
 'denied'             => 134,
 'detected'           => 135,
 'detector'           => 136,
 'device'             => 137,
 'dial'               => 138,
 'dialing'            => 139,
 'dim'                => 140,
 'dining_room'        => 141,
 'disable'            => 142,
 'disarm'             => 143,
 'disarmed'           => 144,
 'dock'               => 145,
 'door'               => 146,
 'doors'              => 147,
 'down'               => 148,
 'driveway'           => 149,
 'east'               => 150,
 'emergency'          => 151,
 'enable'             => 152,
 'end'                => 153,
 'energy'             => 154,
 'enrollment'         => 155,
 'enter'              => 156,
 'entering'           => 157,
 'entertainment'      => 158,
 'enter_the'          => 159,
 'entry'              => 160,
 'environment'        => 161,
 'equipment'          => 162,
 'error'              => 163,
 'evacuate'           => 164,
 'event'              => 165,
 'exercise'           => 166,
 'expander'           => 167,
 'exit'               => 168,
 'exterior'           => 169,
 'f'                  => 170,
 'fail'               => 171,
 'failure'            => 172,
 'family_room'        => 173,
 'fan'                => 174,
 'feed'               => 175,
 'fence'              => 176,
 'fire'               => 177,
 'first'              => 178,
 'flood'              => 179,
 'floor'              => 180,
 'followed'           => 181,
 'force'              => 182,
 'fountain'           => 183,
 'foyer'              => 184,
 'freeze'             => 185,
 'front'              => 186,
 'full'               => 187,
 'furnace'            => 188,
 'fuse'               => 189,
 'game'               => 190,
 'garage'             => 191,
 'gas'                => 192,
 'gate'               => 193,
 'glass'              => 194,
 'go'                 => 195,
 'good'               => 196,
 'goodbye'            => 197,
 'great'              => 198,
 'group'              => 199,
 'guest'              => 200,
 'gun'                => 201,
 'hall'               => 202,
 'hallway'            => 203,
 'hanging_up'         => 204,
 'hang_up'            => 205,
 'has'                => 206,
 'has_expired'        => 207,
 'have'               => 208,
 'hear_menu_options'  => 209,
 'heat'               => 210,
 'help'               => 211,
 'high'               => 212,
 'hold'               => 213,
 'home'               => 214,
 'hot'                => 215,
 'hottub'             => 216,
 'house'              => 217,
 'humidity'           => 218,
 'hvac'               => 219,
 'if'                 => 220,
 'immediately'        => 221,
 'in'                 => 222,
 'inches'             => 223,
 'increase'           => 224,
 'inner'              => 225,
 'input'              => 226,
 'inside'             => 227,
 'instant'            => 228,
 'interior'           => 229,
 'in_the'             => 230,
 'intruder'           => 231,
 'intrusion'          => 232,
 'invalid'            => 233,
 'is'                 => 234,
 'is_about_to_expire' => 235,
 'is_active'          => 236,
 'is_armed'           => 237,
 'is_canceled'        => 238,
 'is_closed'          => 239,
 'is_disarmed'        => 240,
 'is_low'             => 241,
 'is_off'             => 242,
 'is_ok'              => 243,
 'is_on'              => 244,
 'is_open'            => 245,
 'jacuzzi'            => 246,
 'jewelry'            => 247,
 'keep'               => 248,
 'key'                => 249,
 'keypad'             => 250,
 'kitchen'            => 251,
 'lamp'               => 252,
 'laundry'            => 253,
 'lawn'               => 254,
 'leak'               => 255,
 'leave'              => 256,
 'left'               => 257,
 'less'               => 258,
 'level'              => 259,
 'library'            => 260,
 'light'              => 261,
 'lights'             => 262,
 'line'               => 263,
 'living_room'        => 264,
 'loading'            => 265,
 'lobby'              => 266,
 'location'           => 267,
 'lock'               => 268,
 'low'                => 269,
 'lower'              => 270,
 'm'                  => 271,
 'machine'            => 272,
 'mail'               => 273,
 'main'               => 274,
 'mains'              => 275,
 'manual'             => 276,
 'master'             => 277,
 'max'                => 278,
 'media'              => 279,
 'medical'            => 280,
 'medicine'           => 281,
 'memory'             => 282,
 'menu'               => 283,
 'message'            => 284,
 'middle'             => 285,
 'minute'             => 286,
 'missing'            => 287,
 'mode'               => 288,
 'module'             => 289,
 'monitor'            => 290,
 'more'               => 291,
 'motion'             => 292,
 'motor'              => 293,
 'next'               => 294,
 'night'              => 295,
 'no'                 => 296,
 'normal'             => 297,
 'north'              => 298,
 'not'                => 299,
 'notified'           => 300,
 'now'                => 301,
 'number'             => 302,
 'nursery'            => 303,
 'of'                 => 304,
 'off'                => 305,
 'office'             => 306,
 'oh'                 => 307,
 'ok'                 => 308,
 'on'                 => 309,
 'online'             => 310,
 'only'               => 311,
 'open'               => 312,
 'operating'          => 313,
 'option'             => 314,
 'or'                 => 315,
 'other'              => 316,
 'out'                => 317,
 'outlet'             => 318,
 'output'             => 319,
 'outside'            => 320,
 'over'               => 321,
 'overhead'           => 322,
 'panel'              => 323,
 'panic'              => 324,
 'parking'            => 325,
 'partition'          => 326,
 'patio'              => 327,
 'pause'              => 328,
 'perimeter'          => 329,
 'personal'           => 330,
 'phone'              => 331,
 'place'              => 332,
 'play'               => 333,
 'please'             => 334,
 'plus'               => 335,
 'pm'                 => 336,
 'police'             => 337,
 'pool'               => 338,
 'porch'              => 339,
 'port'               => 340,
 'pound'              => 341,
 'pounds'             => 342,
 'power'              => 343,
 'press'              => 344,
 'pressure'           => 345,
 'problem'            => 346,
 'program'            => 347,
 'protected'          => 348,
 'pump'               => 349,
 'radio'              => 350,
 'raise'              => 351,
 'ready'              => 352,
 'rear'               => 353,
 'receiver'           => 354,
 'record'             => 355,
 'recreation'         => 356,
 'relay'              => 357,
 'remain_calm'        => 358,
 'remote'             => 359,
 'repeat'             => 360,
 'report'             => 361,
 'reporting'          => 362,
 'reset'              => 363,
 'restored'           => 364,
 'return'             => 365,
 'right'              => 366,
 'roof'               => 367,
 'room'               => 368,
 'running'            => 369,
 'safe'               => 370,
 'save'               => 371,
 'screen'             => 372,
 'second'             => 373,
 'secure'             => 374,
 'security'           => 375,
 'select'             => 376,
 'sensor'             => 377,
 'serial'             => 378,
 'service'            => 379,
 'set'                => 380,
 'setback'            => 381,
 'setpoint'           => 382,
 'setting'            => 383,
 'shed'               => 384,
 'shipping'           => 385,
 'shock'              => 386,
 'shop'               => 387,
 'shorted'            => 388,
 'shunted'            => 389,
 'side'               => 390,
 'silence'            => 391,
 'siren'              => 392,
 'sliding'            => 393,
 'smoke'              => 394,
 'someone'            => 395,
 'south'              => 396,
 'spare'              => 397,
 'speaker'            => 398,
 'sprinkler'          => 399,
 'stairs'             => 400,
 'stairway'           => 401,
 'star'               => 402,
 'start'              => 403,
 'status'             => 404,
 'stay'               => 405,
 'stock'              => 406,
 'stop'               => 407,
 'storage'            => 408,
 'storm'              => 409,
 'studio'             => 410,
 'study'              => 411,
 'sump'               => 412,
 'sun'                => 413,
 'switch'             => 414,
 'system'             => 415,
 'tamper'             => 416,
 'tank'               => 417,
 'task'               => 418,
 'telephone'          => 419,
 'television'         => 420,
 'temperature'        => 421,
 'test'               => 422,
 'thank_you'          => 423,
 'that'               => 424,
 'the'                => 425,
 'theater'            => 426,
 'thermostat'         => 427,
 'third'              => 428,
 'time'               => 429,
 'toggle'             => 430,
 'top'                => 431,
 'transformer'        => 432,
 'transmitter'        => 433,
 'trespassing'        => 434,
 'trouble'            => 435,
 'turn'               => 436,
 'twice'              => 437,
 'type'               => 438,
 'under'              => 439,
 'unit'               => 440,
 'unlocked'           => 441,
 'unoccupied'         => 442,
 'up'                 => 443,
 'user'               => 444,
 'utility'            => 445,
 'vacation'           => 446,
 'valve'              => 447,
 'video'              => 448,
 'violated'           => 449,
 'visitor'            => 450,
 'wake_up'            => 451,
 'walk'               => 452,
 'wall'               => 453,
 'warehouse'          => 454,
 'warning'            => 455,
 'water'              => 456,
 'way'                => 457,
 'welcome'            => 458,
 'west'               => 459,
 'what'               => 460,
 'when'               => 461,
 'where'              => 462,
 'will'               => 463,
 'window'             => 464,
 'windows'            => 465,
 'with'               => 466,
 'work'               => 467,
 'yard'               => 468,
 'year'               => 469,
 'you'                => 470,
 'zone'               => 471,
 'zones'              => 472,
 '[intruder_message]' => 473);

1;

__END__

=head1 VERSION

Version 0.02

=cut

=head1 AUTHOR

James A. Russo <jr@halo3>

=cut
